src/klein/app.py

```diff
@@ -329,12 +330,33 @@ def run(self, host, port, logFile=None):
 
         @param logFile: The file object to log to, by default C{sys.stdout}
         @type logFile: file object
+
+        @param proto: protocol string for endpoint, can be either "tcp" or "ssl"
+        @type proto: str
+
+        @param privateKey: filepath to private key for ssl
+        @type privateKey: str
+
+        @param certificate: filepath to certificate for ssl
+        @type certificate: str
+
+        @param dhParameters: filepath to dh parameters for ssl
+        @type dhParameters: str
         """
         if logFile is None:
             logFile = sys.stdout
 
         log.startLogging(logFile)
-        reactor.listenTCP(port, Site(self.resource()), interface=host)
+        server_string = "{}:{}:interface={}".format(proto, port, host)
+
+        if privateKey and certificate:
+            server_string += ":privateKey={}:certKey={}".format(privateKey,
+                                                                certificate)
+        if dhParameters:
+            server_string += ":dhParameters={}".format(dhParameters)
+
+        endpoint = endpoints.serverFromString(reactor, server_string)
+        endpoint.listen(Site(self.resource()))
```

Comment:
> A single method solution is feasible, but if it's not something that can be agreed upon, couldn't another method be added and then mark the Klein.run method for deprecation?

deprecating methods is always inconvenient for users so its better to avoid that if possible. It is certainly possible to avoid that here so I think this would be right thing to do.  

> With that being said, what would be the problem with having 2 methods?

if something can be done within one method why add two methods? It's better to keep things simple.

I added commit allowing to pass endpoint_description argument here: https://github.com/twisted/klein/pull/113/commits/c90d181e7884c9f8972dc5d4d678e9cb771dc571